app/src/lib/components/PowerTable.svelte

```diff
@@ -21,6 +21,18 @@ export interface Instructs {
     dataComponent?: ComponentType<SvelteComponent>,
 }
 
+export interface Cell {
+    value: string,
+    rowIndex: number,
+    rowId: number,
+    instructs: Instructs,
+}
+
+export const EDIT_CONSTS = {
```

Comment:
I think the greater benefit here is for a client of the library, where you don't want to "maintain" a string yourself. Specifically for the `SUBMISSION_EVENT`. Say I make 4 custom components, I would probably then want to define my own constant for this message name I need to use in my `dispatch()` call. So I think it is much nicer to import from Powertable than define my own.
Would you be ok with just having the `SUBMISSION_EVENT` constant available for users?

Reply:
If in `src/lib/components/PowerTable.svelte` we have this

```js
export const SUBMISSION_EVENT: 'edits-submitted';
```

and later this

```js
<svelte:component this={instruct.edit?.as} {cell} {...instruct.edit?.props} on:edits-submitted={handleSubmittedEdits}/>
```

users have to always use the event name `edits-submitted` for any number of custom components they create. They can't define their own event names because it has to match what `<svelte:component>` inside PowerTable.svelte is listening to. Or maybe I'm missing your point. If that's the case, please let me know.

Reply:
Sorry I didn't explain clearly, let me try again. I will use the new name ,`edit-submit-event`, from the most recent code (previously `edits-submitted`).

Say I am using PowerTable and have components A, B, C, and D I will use for editing cells. In each component `A.svelte`, `B.svelte`, and so on, I will have a function that is always implemented as such:
```typescript
function submitEdits(e: Event) {
    dispatch('edit-submit-event', {
        rowIndex: rowIndex,
        domEvent: e
    });
}
```

My next step would be to create a constant that I could use to replace the hardcoded string, say a`POWERTABLE_EDIT_SUBMIT_EVENT = 'edit-submit-event'` in `'MyConstants.svelte'` file. And then I can update the function in each A,B,C,D component to use the constant and get rid of the harcoded string:
```typescript
import POWERTABLE_EDIT_SUBMIT_EVENT from 'MyConstants.svelte';
function submitEdits(e: Event) {
    dispatch(POWERTABLE_EDIT_SUBMIT_EVENT, {
        rowIndex: rowIndex,
        domEvent: e
    });
}
```

Instead of having me, the client, define my own constant, it would be nice if PowerTable provided such a constant. So the import becomes `import POWERTABLE_EDIT_SUBMIT_EVENT from '@muonw/powertable';`.

Reply:
The first code is simpler but the second code is easier to maintain. Since there are other strings that users have to hard code (e.g. `rowIndex` key in the code above) and are similarly unlikely to change before a breaking change, hard coding this event name is fine.